### PR TITLE
Create Docker image

### PR DESCRIPTION
**Pull request by marekfilip**
### Description of the Change

I add a Dockerfile requested by issue #10.

### Why Should This Be In The Package?

It fills the assumptions from #10.

### Benefits

User no longer needs to install/compile app by himself. Application is ready to use immediately.

### Possible Drawbacks

There was no application code changes.

### Verification Process

To verify Dockerfile you need to:

1. change directory, where the Dockerfile is

2. build it `docker build --tag vinyldns:vinyldns-cli .`

3. use it, in example:
```
docker run -e VINYLDNS_HOST="https://some-vinyldns.com" \
  -e VINYLDNS_ACCESS_KEY="123" \
  -e VINYLDNS_SECRET_KEY="456" \
  vinyldns/vinyldns-cli \
  zones
```

### Applicable Issues

I would like to point out that I did not use Makefile for compilation. I am not sure if there is there any sense to compile many releases for docker container if we want to use only one of them. Additionally `scratch` image requires `CGO_ENABLED=0 GOOS=linux GOARCH=amd64` build options to run properly (we had to add one more release to Makefile).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM golang:latest AS builder
+
+RUN go get github.com/vinyldns/vinyldns-cli \
```
Comment (mdb) [inline Dockerfile]: This is awesome! Thanks @marekfilip ! I think I'd prefer the `Dockerfile` use the `Makefile` tasks in building `vinyldns` such that the `Makefile` becomes the single source of truth for building `vinyldns`. How do you feel about that?

Comment (marekfilip) [reply Dockerfile]: Not a problem, will do! Image build time will be bit longer, thankfully it's Go so it's not a big deal.

**Comment by marekfilip** [comment]
So, essentially what I've done is an additional build release in Makefile (for Docker scratch image we need to turn CGO off). The bigger change is in Dockerfile.

I couldn't use `go get` this repository because there would be no changes in Makefile therefore there would be no `CGO_ENABLED=0` build. I had to clone my fork to make things work.

```
git clone -b docker-file --single-branch \
        https://github.com/marekfilip/vinyldns-cli.git
```

What I suggest is one of:

1. add `CGO_ENABLED=0` build in another PR or simply change it, then I would replace `git clone` with `go get` to pull this repository (the most proper way)
2. I can do an "anticipating changes", simply change `git clone` to `go get` this repository so everything would work after merge but not before it (the simplest way)

What do you think about that?

**Comment by marekfilip** [comment]
Thanks @mdb for reviews! I forgot that builders and docker repositories have access to the source code as well, that removes all the problems. In this case `git clone` is unnecessary.
I did the modifications, check them out.
I created a docker hub repository to check everything before commit, feel free to see it as well [marekfilip/vinyldns-cli](https://hub.docker.com/r/marekfilip/vinyldns-cli/) (will get removed after we finish).

P. S. you have to spend more time on this PR than you should, sorry for this.

**Comment by pauljamescleary** [comment]
@marekfilip nice!  @mdb once this gets merged, if it makes sense we can do an official release to the approved vinyldns docker hub where our other images live 🎉

**Comment by mdb** [comment]
@marekfilip Thanks! This looks great.

@pauljamescleary Sounds good. I would like to set up automated Docker hub builds on merge to master.

**Comment by marekfilip** [comment]
@mdb, @pauljamescleary Thanks!

**Pull request merged**